JwlMediaWin/Services/UpdateService.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            try
+            {
+                var current = VersionDetection.GetCurrentVersion();
+                var latest = await VersionDetection.GetLatestAsync(ct).ConfigureAwait(false);
+                if (latest == null)
+                {
+                    Log.Warning("[Update] Could not retrieve latest release info.");
+                    return;
+                }
+
+                if (latest.Version > current)
+                {
+                    Log.Information("[Update] Update available: {Latest} (local {Current})",
+                        latest.Version, current);
+
+                    var res = MessageBox.Show(
+                        $"A new version ({latest.Version}) is available. You have {current}.\n\nInstall now?",
+                        "JwlMediaWin - Update available",
+                        MessageBoxButton.YesNo, MessageBoxImage.Information);
+
+                    if (res == MessageBoxResult.Yes)
+                    {
+                        await InstallLatestAsync(latest, ct).ConfigureAwait(false);
+                    }
+                }
+                else
+                {
+                    Log.Information("[Update] Application is up to date ({Version}).", current);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Warning(ex, "[Update] Failed to check for updates.");
+            }
+        }
+
+        public async Task CheckAndInstallSilentAsync(CancellationToken ct = default)
+        {
+            try
+            {
+                var current = VersionDetection.GetCurrentVersion();
+                var latest = await VersionDetection.GetLatestAsync(ct).ConfigureAwait(false);
+                if (latest == null)
+                {
+                    MessageBox.Show("Unable to retrieve the latest release information at this time.",
+                        "JwlMediaWin", MessageBoxButton.OK, MessageBoxImage.Information);
+                    return;
+                }
+
+                if (latest.Version > current)
+                {
+                    await InstallLatestAsync(latest, ct).ConfigureAwait(false);
+                }
+                else
+                {
+                    MessageBox.Show($"You already have the latest version ({current}).",
+                        "JwlMediaWin", MessageBoxButton.OK, MessageBoxImage.Information);
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Warning(ex, "[Update] Silent update flow failed.");
+            }
+        }
+
+        private static async Task InstallLatestAsync(VersionDetection.LatestInfo info, CancellationToken ct)
+        {
+            if (info.AssetUri == null)
+            {
+                Log.Warning("[Update] No downloadable asset found. Opening Releases page instead.");
+                Process.Start(info.HtmlPage.AbsoluteUri);
```

Comment:
Use ProcessStartInfo { UseShellExecute = true } to reliably open URLs on .NET Framework.